src/Controller/ShowProductAction.php

```diff
@@ -27,7 +34,8 @@ public function __invoke(Request $request, string $identifier): Response
     {
         try {
             $locale = $this->guessCurrentLocaleQuery->guess();
-            $product = $this->fetchProductQuery->fetch($identifier, $locale);
+            $catalog = $this->getDefaultCatalog();
+            $product = $this->fetchProductQuery->fetch($catalog->id, $identifier, $locale);
```

Comment:
Shouldn't you check that the catalog is enabled before getting products?

Reply:
Sure i forgot it, If the catalog is disabled, should i throw CatalogDisabledException or should i create a new exception ? (CatalogDisabledException) ?

Reply:
Good question, we raised to Megane the issue with disabled catalogs when fetching products.
She's thinking about changing how the catalogs/{id}/products endpoint behave when the catalog is disabled.
It will impact your PR so don't change it for now.

Reply:
I used CatalogNotFoundException for now to avoid errors